pennylane/_qubit_device.py

```diff
@@ -1437,6 +1437,8 @@ def circuit(x):
 
         if isinstance(obs, CountsMP):
             # convert samples and outcomes (if using) from arrays to str for dict keys
+            samples = np.array([sample for sample in samples if not np.any(np.isnan(sample))])
```

Comment:
Are there any instances where we have to worry about other interfaces? I added `qml.math.isnan` and `qml.math.any` as part of my postselection PR.

Reply:
Ah nevermind QubitDevice just uses numpy.